File: packages/server-core/src/main/kotlin/org/migor/rich/rss/api/ApiParams.kt
package org.migor.rich.rss.api

object ApiParams {
  const val corrId = "x-corr-id"
  const val nonce = "nonce"
}

object WebToFeedParams {
  const val url = "u"
  const val linkPath = "l"
  const val extendContext = "ec"
  const val contextPath = "cp"
  const val paginationXPath = "pp"
  const val datePath = "dp"
  const val prerender = "p"
  const val prerenderWaitUntil = "aw"
  const val filter = "q"
  const val version = "v"
  const val format = "f"
  const val articleRecovery = "ar"
  const val prerenderScript = "ps"
  const val strictMode = "sm"
  const val eventFeed = "ef"
}

object WebToPageChangeParams {
  const val url = "u"
  const val prerender = "p"
  const val prerenderWaitUntil = "aw"
  const val version = "v"
  const val type = "t"
  const val format = "f"
  const val xpath = "x"
  const val prerenderScript = "ps"
}

File: packages/app-web/src/app/components/api-params.ts
export interface WebToFeedParams {
  url: string;
  linkPath: string;
  extendContent: string;
  contextPath: string;
  paginationPath: string;
  datePath: string;
  prerender: string;
  prerenderWaitUntil: string;
  filter: string;
  version: string;
  format: string;
  articleRecovery: string;
  strictMode: string;
  eventFeed: string;
}

export const webToFeedParams: WebToFeedParams = {
  url: 'u',
  linkPath: 'l',
  extendContent: 'ec',
  contextPath: 'cp',
  paginationPath: 'pp',
  datePath: 'dp',
  prerender: 'p',
  prerenderWaitUntil: 'aw',
  filter: 'q',
  version: 'v',
  format: 'f',
  articleRecovery: 'ar',
  strictMode: 'sm',
  eventFeed: 'ef',
};

export interface WebToPageChangeParams {
  url: string;
  prerender: string;
  prerenderWaitUntil: string;
  version: string;
  type: string;
  format: string;
  xpath: string;
}

export const webToPageChangeParams: WebToPageChangeParams = {
  url: 'u',
  prerender: 'p',
  prerenderWaitUntil: 'aw',
  version: 'v',
  type: 't',
  format: 'f',
  xpath: 'x',
};
